pipeline/tick/from_test.go

```diff
@@ -0,0 +1,67 @@
+package tick_test
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/influxdata/kapacitor/tick/ast"
+)
+
+func TestFrom(t *testing.T) {
+	pipe, _, from := StreamFrom()
+	from.Where(&ast.LambdaNode{
+		Expression: &ast.BinaryNode{
+			Operator: ast.TokenAnd,
+			Left: &ast.LambdaNode{
+				Expression: &ast.BinaryNode{
+					Left: &ast.ReferenceNode{
+						Reference: "cpu",
+					},
+					Right: &ast.StringNode{
+						Literal: "cpu-total",
+					},
+					Operator: ast.TokenNotEqual,
+				},
+			},
+			Right: &ast.LambdaNode{
+				Expression: &ast.BinaryNode{
+					Left: &ast.ReferenceNode{
+						Reference: "host",
+					},
+					Right: &ast.RegexNode{
+						Literal: `logger\d+`,
+					},
+					Operator: ast.TokenRegexEqual,
+				},
+			},
+		},
+	})
+	from.GroupBy("this", "that", "these", "those").GroupByMeasurement()
+	from.Database = "mydb"
+	from.RetentionPolicy = "myrp"
+	from.Measurement = "mymeasurement"
+	from.Truncate = time.Second
+	from.Round = time.Second
+
+	got, err := PipelineTick(pipe)
+	if err != nil {
+		t.Fatalf("Unexpected error building pipeline %v", err)
+	}
+
+	want := `stream
+    |from()
+        .database('mydb')
+        .retentionPolicy('myrp')
+        .measurement('mymeasurement')
+        .groupByMeasurement()
+        .round(1s)
+        .truncate(1s)
+        .where(lambda: lambda: "cpu" != 'cpu-total' AND lambda: "host" =~ /logger\d+/)
+        .groupBy('this', 'that', 'these', 'those')
+`
+	if got != want {
+		t.Errorf("TestFrom = %v, want %v", got, want)
+		fmt.Println(got) // print is helpful to get the correct format.
+	}
```

Comment:
I think it would be valuable to wrap this test logic in a helper method. Something like 

```
PipelineTickTestHelper(t *testing.T, pipeline *Pipeline, want string) {
   t.Helper() //mark test as helper so its excluded from the test stack trace.
   got, err := PipelineTick(pipe)
   if err != nil {
       t.Fatalf("Unexpected error building pipeline %v", err)
   }

      if got != want {
         t.Errorf("TestFrom = %v, want %v", got, want)
         fmt.Println(got) // print is helpful to get the correct format.
     }
}
```

Reply:
Does using `t.Log()` instead of `fmt.Println` here server the same purpose? If so that I recommend we use `t.Log()` as it will ensure the output is correctly organized with the test PASS/FAIL output.